agimus_demos_common/launch/franka_common.launch.py

```diff
@@ -0,0 +1,235 @@
+) -> list[LaunchDescriptionEntity]:
+    arm_id = LaunchConfiguration("arm_id")
+    robot_ip = LaunchConfiguration("robot_ip")
+    use_gazebo = LaunchConfiguration("use_gazebo")
+    franka_controllers_params = LaunchConfiguration("franka_controllers_params")
+    linear_feedback_controller_params = LaunchConfiguration(
+        "linear_feedback_controller_params"
+    )
+    use_rviz = LaunchConfiguration("use_rviz")
+    rviz_config_path = LaunchConfiguration("rviz_config_path")
+    gz_verbose = LaunchConfiguration("gz_verbose")
+    gz_headless = LaunchConfiguration("gz_headless")
+
+    robot_ip_empty = context.perform_substitution(robot_ip) == ""
+    use_gazebo_bool = context.perform_substitution(use_gazebo).lower() == "true"
+    if robot_ip_empty and not use_gazebo_bool:
+        raise RuntimeError(
+            "Incorrect launch configuration! Set `robot_ip` to configure hardware or "
+            "`gazebo:=true` to configure simulation."
+        )
+
+    if not robot_ip_empty and use_gazebo_bool:
+        raise RuntimeError(
+            "Incorrect launch configuration! Can not launch demo with both "
+            "`gazebo:=true` and non-empty `robot_ip`."
+        )
+
+    franka_hardware_launch = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [
+                PathJoinSubstitution(
+                    [
+                        FindPackageShare("agimus_demos_common"),
+                        "launch",
+                        "franka_hardware.launch.py",
+                    ]
+                )
+            ]
+        ),
+        launch_arguments={
+            "robot_ip": robot_ip,
+            "arm_id": arm_id,
+            "franka_controllers_params": franka_controllers_params,
+        }.items(),
+        condition=UnlessCondition(use_gazebo),
+    )
+
+    franka_simulation_launch = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            [
+                PathJoinSubstitution(
+                    [
+                        FindPackageShare("agimus_demos_common"),
+                        "launch",
+                        "franka_simulation.launch.py",
+                    ]
+                )
+            ]
+        ),
+        launch_arguments={
+            "gz_verbose": gz_verbose,
+            "gz_headless": gz_headless,
+        }.items(),
+        condition=IfCondition(use_gazebo),
+    )
+
+    load_linear_feedback_controller = generate_load_controller_launch_description(
+        controller_name="linear_feedback_controller",
+        controller_params_file=linear_feedback_controller_params,
+        extra_spawner_args=["--controller-manager-timeout", "1000"],
+    )
+
+    load_joint_state_estimator = generate_load_controller_launch_description(
+        controller_name="joint_state_estimator",
+        controller_params_file=linear_feedback_controller_params,
+        extra_spawner_args=["--controller-manager-timeout", "1000"],
+    )
```

Comment:
I would create a separate function that you can include for these?

Reply:
How about something like `franka_common_lfc.llaunch.py` for this? I also thought of a helper, but maybe simply launch file without LFC and one that extends it with LFC?

Reply:
Maybe that can be good yes.
At least it's obvious from the user side that there is the LFC running when you read the main launch file.
I don't have a strong opinion on the launch file vs helper function.